test/e2e/store_gateway_test.go

```diff
@@ -471,8 +471,8 @@ func TestStoreGatewayNoCacheFile(t *testing.T) {
 		testutil.Ok(t, objstore.UploadDir(ctx, l, bkt, path.Join(dir, id5.String()), id5.String()))
 
 		// Wait for store to sync blocks.
-		// thanos_blocks_meta_synced: 2x loadedMeta 1x labelExcludedMeta 1x TooFreshMeta 1x noMeta.
-		testutil.Ok(t, s1.WaitSumMetrics(e2emon.Equals(5), "thanos_blocks_meta_synced"))
+		// thanos_blocks_meta_synced: 2x loadedMeta 1x labelExcludedMeta 1x.
+		testutil.Ok(t, s1.WaitSumMetrics(e2emon.Equals(4), "thanos_blocks_meta_synced"))
```

Comment:
Why did this change? :thinking: Also, the comment above this line seems to be missing a part.

Reply:
Good catch, looks like it's a leftover artefact of my debugging. I reverted the changes and the test is still passing.

Reply:
How did it work before? :joy: this should fail with `4`. What the hell :smile:

Reply:
I think it's because it waits for the sum to become equal to the asserted value. So in theory it should fail if we set it to `6`, but anything bellow `5` should pass with enough retries :)